Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/services/impl/CertificateManagerServiceImpl.java

```diff
@@ -269,4 +271,87 @@ public boolean deleteCertificateFromTrustStore(String alias) throws
     public boolean validateTrustStorePassKey(String passkey) {
         return cmHelper.getTrustStoreSystemProperties().get(TRUST_STORE_PASSWORD_KEY).equals(passkey);
     }
+
+    @Override
+    public boolean updateCertificateTS(String oldAlias, Certificate cert) throws CertificateManagerException {
+        final Map<String, String> trustStoreProperties = cmHelper.getTrustStoreSystemProperties();
+        String storeType = trustStoreProperties.get(TRUST_STORE_TYPE_KEY);
+        final String storeLoc = trustStoreProperties.get(TRUST_STORE_KEY);
+        final String passkey = trustStoreProperties.get(TRUST_STORE_PASSWORD_KEY);
+        FileInputStream is = null;
+        FileOutputStream os = null;
+        boolean isUpdateSuccessful = false;
+        try {
+            KeyStore tstore = KeyStore.getInstance(storeType);
+            if (!PKCS11_TYPE.equalsIgnoreCase(storeType)) {
+                is = new FileInputStream(storeLoc);
+            }
```

Comment:
Factor out the common logic in a private method for both updateCertificateTS and updateCertificateKS